src/components/chart/chart.tsx

```diff
@@ -422,4 +433,38 @@ export class Chart {
     private recalculateRangeData() {
         this.range = this.calculateRange();
     }
+
+    private handleClick = (event: MouseEvent) => {
+        const target = event.currentTarget as HTMLElement;
+        const index = target.dataset.index;
+        if (index === undefined) {
+            return;
+        }
+
+        const item = this.items[Number(index)];
+        if (!item.clickable) {
+            return;
+        }
+
+        event.stopPropagation();
+        this.itemSelected.emit(item);
+    };
+
+    private handleKeyDown = (event: KeyboardEvent) => {
+        const target = event.currentTarget as HTMLElement;
+        const index = target.dataset.index;
+        if (index === undefined) {
+            return;
+        }
+
+        const item = this.items[Number(index)];
+        if (!item.clickable) {
+            return;
+        }
+
+        if (event.key === 'Enter' || event.key === ' ') {
+            event.preventDefault();
+            this.handleClick(event as unknown as MouseEvent);
```

Comment:
This forced recasting of a `KeyboardEvent` to a `MouseEvent` means that someone might later make changes in the `handleClick` function, which causes no compiler warnings, and works just fine when tested with clicks, but which throws an uncaught error bringing the app, or parts of it, down at runtime.

So I'm opposed 😝

Either break out the parts that you need into a separate function which are called by both `handleKeyDown` and `handleClick`, or update `handleClick` to accept either kind of event:

```ts
private handleClick = (event: MouseEvent | KeyboardEvent) => {
```

Although, in the latter case, it does look kind of weird that `handleClick` accepts a keyboard event, so we might consider renaming the function in that case.

Reply:
I swear to my imaginary entity that I had done this, but forgot to push it. 🙈

Now ⚡